ai-service/app/ml/cancellation/synthetic.py

```
@@ -0,0 +1,100 @@
+    Returns:
+        bookings: list of booking dicts matching Spring Boot features format
+        labels: list of binary labels (1=cancelled, 0=not cancelled)
+    """
+    random.seed(seed)
+    bookings = []
+    labels = []
+
+    n_cancelled = int(n * cancel_ratio)
+
+    for i in range(n):
+        prop_type = random.choice(PROPERTY_TYPES)
```

Ruff (0.16.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)